framework/src/systems/NonlinearSystem.C

```diff
@@ -512,8 +512,8 @@ NonlinearSystem::computeScalingJacobian()
     petsc_matrix.reset_preallocation();
 
 #else
-    mooseWarning("Automatic scaling requires a PETSc version of 3.9.0 or greater, so no automatic "
-                 "scaling is going to be performed");
+    mooseError("Automatic scaling requires a PETSc version of 3.9.0 or greater, so no automatic "
```

Comment:
paramError?

Reply:
The only reason I like param error over regular error is it makes it very clear for the user exactly where it comes with line numbers.

Reply:
`NonlinearSystem` is not a `MooseObject` so we cannot do `paramError`. However, I've added a check to the `Executioner` constructor that ***will*** generate a `paramError` if automatic scaling has been requested and we don't have the requisite PETSc version.

Reply:
Not that you need to here, but the actual information for making nice paramError-like messages is stored on the InputParameters object itself.  So you could just follow the pattern of code in the paramError implementation that uses that info to make an error message following the same pattern.

Reply:
Yea but `NonlinearSystem` doesn't have an `InputParameters` object